src/services/users_service.ts

```diff
@@ -1,9 +1,25 @@
 import crypto from 'crypto';
-// import { User } from '../models';
+import db from '../models';
+import User from '../models/users_model';
 
-export const signup = async () => {
-  try {
-  } catch (err) {}
+export const signup = async (email: string, name: string, password: string) => {
+  const salt = crypto.randomBytes(64).toString('base64');
+  const hashedPassword = hashPassword(password, salt);
+  const user = await db.User.create({
+    email,
+    name,
+    password: hashedPassword,
+    passwordSalt: salt,
+  });
+  return user;
+};
+
+export const signin = (user: User, password: string) => {
+  const hashedPassword = hashPassword(password, user.passwordSalt);
+  if (hashedPassword != user.password) {
+    return false;
+  }
+  return true;
```

Comment:
```suggestion
  return hashedPassword;
```
해서 controller에서 
```javascript
  if (hashedPassword != user.password) {
    return res.status(Bad_request).json(비밀번호가 맞지 않음);
  }
  // ... 이후 로그인 로직 수행
```
이 더 괜찮지 않을까? 
지금 코드는 분기처리를 굳이 Service에서 한번하고 controller에서 한번 더하는 느낌?

Reply:
일단은 이렇게하고 다음에 더 리팩토링하면서 고쳐볼게요 ~